### ISSUE TITLE
Inline arbitrary definition

### ISSUE THREAD

**Issue by kaikoga** [ArpEngine/ArpDomain#29]
~~I think this is blocked by #30~~

## Background

```
<obj objRefWithSomeWeirdName="ref1" objRefWithOtherWeirdName="ref2">
</obj>
```

would be inline defined like this

```
<obj>
  <objRefWithSomeWeirdName value="1" />
  <objRefWithOtherWeirdName value="2" />
</obj>
```

but this would be cleaner:

```
<obj objRefWithSomeWeirdName="ref1" objRefWithOtherWeirdName="ref2">
  <obj name="objRefWithSomeWeirdName" value="1" />
  <obj name="objRefWithOtherWeirdName" value="2" />
</obj>
```

though it is impossible because currently `<obj>` cannot accept `<obj>` seed

**Comment by kaikoga** [comment]
Just allow `<data>` inside anything?

**Comment by kaikoga** [comment]
Maybe named anything could be allowed, but we should consider checking slot overwrite

**Comment by kaikoga** [comment]
Going to low priority, because we can write this:
```
<data>
  <obj name="@" objRefWithSomeWeirdName="ref1" objRefWithOtherWeirdName="ref2" />
  <obj name="objRefWithSomeWeirdName" value="1" />
  <obj name="objRefWithOtherWeirdName" value="2" />
</data>
```
